core/mlang2.js

```diff
@@ -956,9 +964,13 @@ CLASS({
       var acc = this.arg2;
       acc.put(val);
     },
-    clone: function() {
+    exprClone: function() {
       // Don't use default clone because we don't want to copy 'groups'
-      return MapExpr.create({arg1: this.arg1, arg2: this.arg2.clone()});
+      return MapExpr.create({
+        arg1: this.arg1,
+        arg2: Array.isArray(this.arg2) ? this.arg2.sink.exprClone() :
```

Comment:
If the user puts [].dao as arg2 and you turn it into [].sink on clone then you're misrepresenting what the user wanted.